EiDiA_frontend/src/components/record/Widgets/Widget.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import {Centering, FoggyDiv, H2WithOutMargin} from "../../StyleElements";
+import {FiEdit2} from "react-icons/fi";
+import Fab from "@material-ui/core/Fab";
+import {DragTypes} from "../../../assets/Constants";
+import {EditDialog} from "./EditDialog";
+import {DragSource} from "react-dnd";
+/**
+ * TODO:
+ *
+ */
+
+/*
+ *Reason for no use of inheritance as specified in Data model, instead composition is used
+ * https://reactjs.org/docs/composition-vs-inheritance.html#so-what-about-inheritance
+ */
+
+
+const itemSource = {
+    canDrag(props) {
+        return props.dashboardEditingActive;
+    },
+    beginDrag(props) {
+        return {positionInfo: props.widget.positionInfo, switchWidget: props.switchWidget}
+
+    },
+    endDrag(props, monitor) {
+        if (monitor.getDropResult() == null) return;
+        let wrapperProps = monitor.getDropResult().component.props
+        let draggedItem = monitor.getItem()
+
+        draggedItem.switchWidget(wrapperProps.positionInfo, draggedItem.positionInfo)
+    }
+}
+
+function collectDrag(connect, monitor) {
+    return {
+        connectDragSource: connect.dragSource(),
+        isDragging: monitor.isDragging()
+    }
+}
+
+
+class Widget extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            widget: this.props.widget,
+            widgetEditingActive: false,
+            dashboardEditingActive: this.props.dashboardEditingActive,
+            editingActive: false
+        }
+
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (prevProps !== this.props) {
+            this.setState({
+                widget: this.props.widget,
+                dashboardEditingActive: this.props.dashboardEditingActive
+
+            })
+        }
+    }
+
+    toggleEditDialog = () => {
+        this.setState({widgetEditingActive: !this.state.widgetEditingActive})
+
+    }
+
+    MainPart() {
```

Comment:
```suggestion
    renderMainPart() {
```

und dann genauso unten angepasst